File: src/main/java/com/chatgpt/example/controller/ChatGPTRestController.java
package com.chatgpt.example.controller;

import com.chatgpt.example.dto.request.GPTCompletionChatRequest;
import com.chatgpt.example.dto.response.CompletionChatResponse;
import com.chatgpt.example.dto.response.CompletionResponse;
import com.chatgpt.example.dto.request.GPTCompletionRequest;
import com.chatgpt.example.service.GPTChatRestService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/chatgpt/rest")
@RequiredArgsConstructor
public class ChatGPTRestController {

  private final GPTChatRestService gptChatRestService;

  @PostMapping("/completion")
  public CompletionResponse completion(final @RequestBody GPTCompletionRequest gptCompletionRequest) {

    return gptChatRestService.completion(gptCompletionRequest);
  }

  @PostMapping("/completion/chat")
  public CompletionChatResponse completionChat(final @RequestBody GPTCompletionChatRequest gptCompletionChatRequest) {

    return gptChatRestService.completionChat(gptCompletionChatRequest);
  }

}

File: chatGPT_example_kotlin/src/main/kotlin/com/example/chatgpt_example_kotlin/controller/ChatGPTRestController.kt
package com.example.chatgpt_example_kotlin.controller

import com.example.chatgpt_example_kotlin.dto.CompletionChatDTO
import com.example.chatgpt_example_kotlin.dto.CompletionChatResponseDTO
import com.example.chatgpt_example_kotlin.dto.CompletionRequestDTO
import com.example.chatgpt_example_kotlin.dto.CompletionResponseDTO
import com.example.chatgpt_example_kotlin.service.GPTChatService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/chat")
class ChatGPTRestController(
        private val gptChatService: GPTChatService
) {

    @PostMapping("/completion")
    fun completion(
            @RequestBody completionRequestDTO: CompletionRequestDTO
    ): CompletionResponseDTO {
        return gptChatService.completion(completionRequestDTO)
    }

    @PostMapping("/completion/chat")
    fun completionChat(
            @RequestBody completionChatRequestDTO: CompletionChatDTO,
    ): CompletionChatResponseDTO {
        return gptChatService.completionChat(completionChatRequestDTO)
    }
}